include/triton-shared/Dialect/TritonStructured/IR/TritonStructuredDialect.td

```diff
@@ -0,0 +1,137 @@
+#ifndef TRITON_STRUCTURED_DIALECT
+#define TRITON_STRUCTURED_DIALECT
+
+include "mlir/IR/OpBase.td"
+include "triton/Dialect/Triton/IR/TritonTypes.td"
+include "mlir/Interfaces/SideEffectInterfaces.td" // Pure
+
+def Triton_Structured_Dialect : Dialect {
+  let name = "tts";
+
+  let cppNamespace = "::mlir::tts";
+
+  let summary = "Structured Triton operations";
+
+  let description = [{
+    Triton Structured Dialect.
+  }];
+
+  let dependentDialects = [
+    "triton::TritonDialect"
+  ];
+
+  let usePropertiesForAttributes = 1;
+}
+
+//
+// Op Base
+//
+class TTS_Op<string mnemonic, list<Trait> traits = []> :
+    Op<Triton_Structured_Dialect, mnemonic, traits> {
+}
+
+def TTS_MakeTensorPtrOp
+  : TTS_Op<"make_tptr", [ AttrSizedOperandSegments, Pure]> {
+  let summary = "create a pointer that points to a tensor in memory";
+
+  let arguments = (ins TT_Ptr:$base,
+                       DenseI64ArrayAttr:$sizes,
+                       Variadic<Index>:$strides,
+                       Variadic<Index>:$offsets,
+                       Variadic<Index>:$parent_sizes,
```

Comment:
I think the other variables are self-explanatory, but for `parent_sizes`, it might be worth mentioning that it's for tracking the sizes of the parent used in a modulo expression

Reply:
Added explanation on each field on top.